c/tskit/tables.c

```diff
@@ -10272,6 +10272,7 @@ tsk_table_collection_check_tree_integrity(const tsk_table_collection_t *self)
     const tsk_id_t *restrict edge_child = self->edges.child;
     const tsk_id_t *restrict edge_parent = self->edges.parent;
     tsk_id_t *restrict parent = NULL;
+    int *restrict used_edges = NULL;
```

Comment:
since this can only take values 0, 1 or 2 I think we should save a bit of memory and make it an ``int8_t``.